src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -846,45 +846,29 @@ static int enumerate_partitions(dev_t devnum) {
                 return ok ? 0 : r;
         }
 
-        if (m->partitions[PARTITION_SWAP].found) {
-                k = add_partition_swap(m->partitions + PARTITION_SWAP);
-                if (k < 0)
-                        r = k;
-        }
+        if (m->partitions[PARTITION_SWAP].found)
+                RET_GATHER(r, add_partition_swap(m->partitions + PARTITION_SWAP));
 
-        k = process_loader_partitions(m->partitions + PARTITION_ESP, m->partitions + PARTITION_XBOOTLDR);
-        if (k < 0)
-                r = k;
+        RET_GATHER(r, process_loader_partitions(m->partitions + PARTITION_ESP, m->partitions + PARTITION_XBOOTLDR));
```

Comment:
Please also update `process_loader_partitions` to make it return the first error.

Reply:
Updated.

Reply:
Updated.